### PR TITLE
tests: add TestBackEnd AIO and NULL Get and Stats

### PR DESCRIPTION
**Pull request by glimchb**
Signed-off-by: Boris Glimcher <Boris.Glimcher@emc.com>

### CODE REVIEW

File: pkg/backend/aio_test.go [hunk 1]
```diff
@@ -224,12 +228,203 @@ func TestBackEnd_ListAioControllers(t *testing.T) {
...
+		start   bool
+	}{
+		{
+			"valid request with invalid SPDK response",
+			"volume-test",
+			nil,
+			[]string{`{"id":%d,"error":{"code":0,"message":""},"result":[]}`},
+			codes.InvalidArgument,
+			fmt.Sprintf("expecting exactly 1 result, got %v", "0"),
+			true,
```
Comment (artek-koltun) [inline pkg/backend/aio_test.go:248]: Do we need to set a member to true in all cases? Can we just pass true directly to `createTestEnvironment`?

File: pkg/backend/aio_test.go [hunk 2]
```diff
@@ -224,12 +228,203 @@ func TestBackEnd_ListAioControllers(t *testing.T) {
...
+	// run tests
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			testEnv := createTestEnvironment(tt.start, tt.spdk)
+			defer testEnv.Close()
+
+			request := &pb.GetAioControllerRequest{Name: tt.in}
+			response, err := testEnv.client.GetAioController(testEnv.ctx, request)
+			if response != nil {
+				// Marshall the request and response, so we can just compare the contained data
```
Comment (artek-koltun) [inline pkg/backend/aio_test.go:306]: do we need such obvious comments? The code is not convoluted to understand what we are doing here...

File: pkg/backend/aio_test.go [hunk 3]
```diff
@@ -224,12 +228,203 @@ func TestBackEnd_ListAioControllers(t *testing.T) {
...
+				mtt, _ := proto.Marshal(tt.out)
+				mResponse, _ := proto.Marshal(response)
+				// Compare the marshalled messages
+				if !bytes.Equal(mtt, mResponse) {
+					t.Error("response: expected", tt.out, "received", response)
+				}
+			}
+
+			if err != nil {
```
Comment (artek-koltun) [inline pkg/backend/aio_test.go:315]: What if we expect an error and got `nil`? Probably we should fail then...

**Comment by glimchb** [comment]
@artek-koltun all great comments. 
But commonality across tests is more important for me now.
If/When we going to make a change it should be across all tests
you already opened #265 to track this

**Comment by glimchb** [comment]
> For some cases, this is ok. However, in my opinion, we just add maintainability burden with some tests, which just test some spdk json rpc module behavior, but not business logic of an opi call

I agree. One of reasons I suggested to have a sperate package for `gospdk` ...

**Pull request merged**